pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -383,6 +383,14 @@ let
 
   intermediatesDir = "share/haskell/${ghc.version}/${pname}-${version}/dist";
 
+  # On old ghcjs, the jsexe directories are the output but on the js backend they seem to be treated as intermediates
```

Comment:
Not sure this was intended, nor whether `-DGHCJS_BROWSER` is meant to change that behavior
Thoughts @luite  ?